playwright/tests/customers.spec.ts

```diff
@@ -0,0 +1,234 @@
+import { ADDRESS } from "@data/addresses";
+import { CUSTOMERS } from "@data/e2eTestData";
+import { AddressesListPage } from "@pages/addressesListPage";
+import { CustomersPage } from "@pages/customersPage";
+import { AddAddressDialog } from "@pages/dialogs/addAddressDialog";
+import { DeleteAddressDialog } from "@pages/dialogs/deleteAddressDialog";
+import { AddressForm } from "@pages/forms/addressForm";
+import { GiftCardsPage } from "@pages/giftCardsPage";
+import { expect, test } from "@playwright/test";
+import faker from "faker";
+
+test.use({ storageState: "./playwright/.auth/admin.json" });
+
+let customersPage: CustomersPage;
+let giftCardsPage: GiftCardsPage;
+let addressesListPage: AddressesListPage;
+let addressForm: AddressForm;
+let deleteAddressDialog: DeleteAddressDialog;
+let addAddressDialog: AddAddressDialog;
+
+test.beforeEach(({ page }) => {
+  customersPage = new CustomersPage(page);
+  giftCardsPage = new GiftCardsPage(page);
+  addressesListPage = new AddressesListPage(page);
+  addressForm = new AddressForm(page);
+  addAddressDialog = new AddAddressDialog(page);
+  deleteAddressDialog = new DeleteAddressDialog(page);
+});
+
+test("TC: SALEOR_199 Create customer @e2e @customer", async () => {
+  const firstName = faker.name.firstName();
+  const lastName = faker.name.lastName();
+  const note = faker.lorem.sentence();
+  const email = faker.internet.email();
+
+  await customersPage.goToCustomersListView();
+  await customersPage.clickOnCreateCustomer();
+  await customersPage.fillFirstAndLastName(firstName, lastName);
+  await customersPage.fillEmail(email);
+
+  const newAddress = ADDRESS.addressUS;
```

Comment:
Most of this data is the same as in previous tests, we have a lot of duplications here. Can we separate common areas and reuse them using DRY principles? For example Customer Create and address create could be a helper, or filling address could be in page object


// Helper function for customer creation steps

async function createCustomer(firstName: string, lastName: string, email: string, address: typeof ADDRESS.addressUS)
 {
  await customersPage.goToCustomersListView();
  await customersPage.clickOnCreateCustomer();
  await customersPage.fillFirstAndLastName(firstName, lastName);
  await customersPage.fillEmail(email);
  await addressForm.completeAddressForm(address);
  await customersPage.fillNote(faker.lorem.sentence());
  await customersPage.saveCustomer();

Reply:
It will be handled in the next PR